src/shared/listener-collection.js

```diff
@@ -11,24 +11,41 @@
  */
 export class ListenerCollection {
   constructor() {
-    /** @type {Listener[]} */
+    /** @type {Array<Listener|null>} */
```

Comment:
There is currently a listener ID reuse hazard in `removeAll` (see comments for that method). Although you could fix that while continuing to use an array, I think it would be more natural to use a map of listener ID to listener, together with an ID counter that doesn't get reset.